clusters/resource_cluster.go

```diff
@@ -234,21 +221,60 @@ func fixInstancePoolChangeIfAny(d *schema.ResourceData, cluster *Cluster) {
 	}
 }
 
+func listClusterConfigChangedKeys(d *schema.ResourceData) []string {
+	changedKeys := []string {}
+	for k := range clusterSchema {
+		// TODO: create a map if we'll add more non-cluster config parameters in the future
+		if k == "library" || k == "is_pinned" {
+			continue
+		}
+		a,b := d.GetChange(k)
+		if d.HasChange(k) {
+			changedKeys = append(changedKeys, k)
+		}
+	}
+	return changedKeys
+}
+
 func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
 	clusters := NewClustersAPI(ctx, c)
 	clusterID := d.Id()
 	cluster := Cluster{ClusterID: clusterID}
 	common.DataToStructPointer(d, clusterSchema, &cluster)
 	var clusterInfo ClusterInfo
 	var err error
-	if hasClusterConfigChanged(d) {
+
+	clusterChangedConfigList := listClusterConfigChangedKeys(d)
+
+	if len(clusterChangedConfigList) > 0 {
 		log.Printf("[DEBUG] Cluster state has changed!")
 		if err := cluster.Validate(); err != nil {
 			return err
 		}
 		cluster.ModifyRequestOnInstancePool()
 		fixInstancePoolChangeIfAny(d, &cluster)
-		clusterInfo, err = clusters.Edit(cluster)
+
+		// We can only call the resize api if the cluster is in the running state
+		// and only the cluster siz (ie num_workers OR autoscale) is being changed
+		hasOnlyClusterSizeChanged := len(clusterChangedConfigList) == 1 && (clusterChangedConfigList[0] == "num_workers" || clusterChangedConfigList[0] == "autoscale")
```

Comment:
How it will behave when you change from `num_workers` to `autoscale` ? won't it be two changes?

Reply:
Yeah you are right. Seems like resize also supports autoscale <-> num_workers. I'll make the necessary changes

Reply:
done